### PR TITLE
koordlet: make UsageHighEnough configurable for cpu_evict

### PR DESCRIPTION
**Pull request by fengyehong**
### Ⅰ. Describe what this PR does

cpu_evict: make UsageHighEnough configurable

### Ⅱ. Does this pull request fix one issue?

NONE

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Review by jasonliu747 — COMMENTED** [review]
I'm fine with this change, but there might be a better variable naming for `CPUEvictBEUsageHighEnough`.

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 1]
```diff
@@ -177,8 +177,10 @@ type ResourceThresholdStrategy struct {
 
 	// if be CPU RealLimit/allocatedLimit > CPUEvictBESatisfactionUpperPercent, then stop evict BE pods
 	CPUEvictBESatisfactionUpperPercent *int64 `json:"cpuEvictBESatisfactionUpperPercent,omitempty"`
-	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage nearly 100%) continue CPUEvictTimeWindowSeconds,then start evict
+	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage >= CPUEvictBEUsageHighEnough) continue CPUEvictTimeWindowSeconds,then start evict
 	CPUEvictBESatisfactionLowerPercent *int64 `json:"cpuEvictBESatisfactionLowerPercent,omitempty"`
+	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage >= CPUEvictBEUsageHighEnough) continue CPUEvictTimeWindowSeconds,then start evict
+	CPUEvictBEUsageHighEnough *int64 `json:"cpuEvictBEUsageHighEnough,omitempty"`
```
Comment (saintube) [inline apis/slo/v1alpha1/nodeslo_types.go:183]: to make the naming more concise, how about `CPUEvictBEUsageHighPercent`

Comment (fengyehong) [reply apis/slo/v1alpha1/nodeslo_types.go:183]: i think HighEnough makes sence, it's ok for me to add suffix Percent making it `CPUEvictBEUsageHighEnoughPercent`

Comment (saintube) [reply apis/slo/v1alpha1/nodeslo_types.go:183]: Alright. WDYT @jasonliu747

Comment (jasonliu747) [reply apis/slo/v1alpha1/nodeslo_types.go:183]: FMPOV, the word `Threshold` would fit the scenario.

Comment (fengyehong) [reply apis/slo/v1alpha1/nodeslo_types.go:183]: `CPUEvictBEUsageThresholdPercent` or `CPUEvictBEUsageThreshold` ?

Comment (jasonliu747) [reply apis/slo/v1alpha1/nodeslo_types.go]: sorry for the late response. `ThresholdPercent` SGTM.

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 2]
```diff
@@ -177,8 +177,10 @@ type ResourceThresholdStrategy struct {
 
 	// if be CPU RealLimit/allocatedLimit > CPUEvictBESatisfactionUpperPercent, then stop evict BE pods
 	CPUEvictBESatisfactionUpperPercent *int64 `json:"cpuEvictBESatisfactionUpperPercent,omitempty"`
-	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage nearly 100%) continue CPUEvictTimeWindowSeconds,then start evict
+	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage >= CPUEvictBEUsageThresholdPercent) continue CPUEvictTimeWindowSeconds,then start evict
 	CPUEvictBESatisfactionLowerPercent *int64 `json:"cpuEvictBESatisfactionLowerPercent,omitempty"`
+	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage >= CPUEvictBEUsageThresholdPercent) continue CPUEvictTimeWindowSeconds,then start evict
+	CPUEvictBEUsageThresholdPercent *int64 `json:"cpuEvictBEUsageHighEnough,omitempty"`
```
Comment (jasonliu747) [inline apis/slo/v1alpha1/nodeslo_types.go:183]: don't forget to update json value as well ;)

**Comment by zwzhang0107** [comment]
this looks good, how about add some UTs for this new config.

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 3]
```diff
@@ -177,8 +177,10 @@ type ResourceThresholdStrategy struct {
 
 	// if be CPU RealLimit/allocatedLimit > CPUEvictBESatisfactionUpperPercent, then stop evict BE pods
 	CPUEvictBESatisfactionUpperPercent *int64 `json:"cpuEvictBESatisfactionUpperPercent,omitempty"`
-	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage nearly 100%) continue CPUEvictTimeWindowSeconds,then start evict
+	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage >= CPUEvictBEUsageThresholdPercent) continue CPUEvictTimeWindowSeconds,then start evict
```
Comment (saintube) [inline apis/slo/v1alpha1/nodeslo_types.go:180]: `... and usage >= CPUEvictBEUsageThresholdPercent / 100`?
same below

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 4]
```diff
@@ -177,8 +177,10 @@ type ResourceThresholdStrategy struct {
 
 	// if be CPU RealLimit/allocatedLimit > CPUEvictBESatisfactionUpperPercent, then stop evict BE pods
 	CPUEvictBESatisfactionUpperPercent *int64 `json:"cpuEvictBESatisfactionUpperPercent,omitempty"`
-	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage nearly 100%) continue CPUEvictTimeWindowSeconds,then start evict
+	// if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent and usage >= CPUEvictBEUsageThresholdPercent / 100) continue CPUEvictTimeWindowSeconds, then start evict
```
Comment (saintube) [inline apis/slo/v1alpha1/nodeslo_types.go:180]: Sorry for not clarifying the problem clearly. We expect to correct the comments related to XXXPercent.

e.g.

```
if be CPU RealLimit/allocatedLimit > CPUEvictBESatisfactionUpperPercent/100
if be CPU (RealLimit/allocatedLimit < CPUEvictBESatisfactionLowerPercent/100 and usage ...
```

Comment (fengyehong) [reply apis/slo/v1alpha1/nodeslo_types.go:180]: what about an individual PR to do that, its not topic related

Comment (fengyehong) [reply apis/slo/v1alpha1/nodeslo_types.go]: updated

**Review by saintube — APPROVED** [review]
/lgtm

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Pull request merged**